[31023] Preparation for project edit fields: Generalize edit and display fields currently work package specific

HDinger opened the pull request:
This PR aims to make all parts that belong to the inplace edit field, independent of WorkPackages. Thus we can use those fields for project attributes as well. 

This includes generalizing

- [x] The edit fields
  - [x] The `edit-fields`-types
  - [x] The actual `edit-field.component` class and subclasses
  - [x] The `wp-edit-field-group.component` to generalized `edit-field-group.component`
  - [x] The `wp-edit-field-handler` and `edit-field-handler` interface
  - [x] The `wp-edit-form` class
  - [x] The `edit-form-portal.component`
  - [x] The `wp-editing-portal` service
  - [x] The `wp-edit-field.component`
- [x] Consolidate EditForm and EditContext
  - [x] Both are joined into one class
  - [x] `EditFieldGroupComponent` is changed into `EditFormComponent`
- [x] The display fields
  - [x] The `display-field`-types
  - [x] The `display-field-renderer`
  - [x] Special handling for display portals (avatars e.g.)
  - [x] Special handling for `highlightable` fields
- [x] The belonging services
  - [x] The wpNotifications service
    - [x] Special handling for notifications in work packages ("Show in fullscreen", "Reload due to conflict" etc)
  - [x] The wpEvents service needs to be generalized for other resources
    - [x] Special handling / filtering of events for specific resource types
  - [x] The wpEditing service
    - [x] Special handling for creating changesets that are NOT `ResourceChangeset<HalResource>` (such as `WorkPackageChangeset`)
  - [x] The wpCreate service
- [x] Nice to have:
  - [x] Consolidate all edit and display field classes from their `wp-` prefixes

Ideally, all those work package specific components can be generalized into a generic `HalResource`-based component and augmented with a hierarchical-injectable service that provides customization, such as saving success state routes and changeset creation etc.
These services would then be created once per editable resource, IF there are any specific handling callbacks to process.

The changeset were opened in a previous [PR](https://github.com/opf/openproject/pull/7677) to handle not only WorkPackages but also other resources.

HDinger committed:
Check edit fields for Work package dependencies

HDinger committed:
Make EditForm, NotificationService, EditContext indepenedent of Work Packages

HDinger committed:
Make EventsService independent of WorkPackages

HDinger committed:
Start renaming WpNotificationService (WiP)

HDinger commented:
https://community.openproject.com/projects/openproject/work_packages/31023/activity

openprojectci commented:
<!--
  0 Errors
  0 Warnings
  0 Messages
  1 Markdown
-->

## Danger has errored
[!] Invalid `Dangerfile` file: No such file or directory @ rb_sysopen - frontend/src/app/components/wp-edit-form/work-package-edit-context.ts
```
 #  from Dangerfile:8
 #  -------------------------------------------
 #      .each do |path|
 >    lines = File.readlines(path)
 #  
 #  -------------------------------------------
```

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

oliverguenther committed:
Fix more references

HDinger committed:
Fix typescript errors

HDinger committed:
Add basic halEditingService

HDinger committed:
Rename to global halResourceEditingService (WiP)

HDinger committed:
Move typing from class to methods (WiP)

oliverguenther committed:
Fix typescript errors

oliverguenther committed:
Remove space in type

HDinger committed:
Add test project widget && generalize the editFieldGroup

oliverguenther committed:
Rename editing portal service

[ci skip]

HDinger committed:
Rename WpEditFieldComponent

HDinger committed:
Rename WpDisplayFields

oliverguenther committed:
Rename display-XX-field to XX-display-field

[ci skip]

HDinger committed:
Add WP specific ID field to distinguish between resources

oliverguenther committed:
Re-add state in work package resource

oliverguenther committed:
Generalize display field renderer

oliverguenther committed:
Rename spent-time to wp-spent-time and fix highlight specifics

[ci skip]

oliverguenther committed:
Actually load the project schema and make field editable

oliverguenther committed:
Merge remote-tracking branch 'origin/dev' into feature/31023-In-place-editing-of-project-description-in-widget

oliverguenther committed:
Make edit-field-group.component an edit-form.component and subclass EditForm

oliverguenther committed:
Remove edit context in favor of specialized EditForm

oliverguenther committed:
Add special cases for work package editing

oliverguenther committed:
Fix edit actions bar

HDinger committed:
Fix codeclimate issues

HDinger committed:
Use WorkPackageNotificationService if necessary

HDinger committed:
Override NotificationService for WPs to allow WP specififc notifications (WiP)

oliverguenther committed:
Correctly provide wpNotification service

Because the ui-router doesn't seem to correctly use the parent element's
injector, we need to provide the wpNotification service not in the
wp-base, but rather the wp-list component as well as in the isolated
query space.

oliverguenther committed:
Allow to filter halEvents for specific resourceTypes (e.g. WorkPackage)

oliverguenther committed:
Remove superfluous cell class constant

oliverguenther committed:
Start renaming selectors for wp-edit-field into generics

oliverguenther committed:
Remove wp-table--cell-span in favor of display field selector

oliverguenther committed:
Consolidate other display and edit field styles

oliverguenther committed:
Provide specialized service for transitions in active edit forms

oliverguenther committed:
Remove superfluous overflowSelector

oliverguenther committed:
Accept that date field contains some work package specifics

oliverguenther committed:
Ignore unreadable files

openprojectci commented:
<!--
  0 Errors
  0 Warnings
  0 Messages
  1 Markdown
-->

## Danger has errored
[!] Invalid `Dangerfile` file: undefined method `tqg` for File:Class
```
 #  from Dangerfile:10
 #  -------------------------------------------
 #  
 >    lines = File.tqg (path)
 #  
 #  -------------------------------------------
```

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

oliverguenther committed:
Provide the changeset for work packages as a hook

Since hal resource editing service is provided per query space, we
cannot register them once (would only work globally)

oliverguenther committed:
Fix dangerfile

openprojectci commented on frontend/src/app/components/modals/wp-destroy-modal/wp-destroy.modal.ts:
```diff
@@ -40,6 +40,7 @@ import {StateService} from '@uirouter/core';
 import {WorkPackageService} from "core-components/work-packages/work-package.service";
 import {BackRoutingService} from "core-app/modules/common/back-routing/back-routing.service";
+import {WorkPackageNotificationService} from "core-app/modules/work_packages/notifications/work-package-notification.service";
 
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/370b05bb31fbf6a37ab5f5ec99ceabbe7040fa15/frontend/src/app/components/modals/wp-destroy-modal/wp-destroy.modal.ts#L44"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

openprojectci commented on frontend/src/app/components/work-packages/wp-breadcrumb/wp-breadcrumb-parent.component.ts:
```diff
@@ -29,8 +29,9 @@
+import {HalResourceNotificationService} from "core-app/modules/hal/services/hal-resource-notification.service";
 import {I18nService} from 'core-app/modules/common/i18n/i18n.service';
+import {WorkPackageNotificationService} from "core-app/modules/work_packages/notifications/work-package-notification.service";
 
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/370b05bb31fbf6a37ab5f5ec99ceabbe7040fa15/frontend/src/app/components/work-packages/wp-breadcrumb/wp-breadcrumb-parent.component.ts#L35"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

openprojectci commented on frontend/src/app/components/wp-custom-actions/wp-custom-actions/wp-custom-action.component.ts:
```diff
@@ -30,13 +30,15 @@
-import {WorkPackageEventsService} from "core-app/modules/work_packages/events/work-package-events.service";
+import {HalEventsService} from "core-app/modules/hal/services/hal-events.service";
+import {WorkPackageNotificationService} from "core-app/modules/work_packages/notifications/work-package-notification.service";
 
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/370b05bb31fbf6a37ab5f5ec99ceabbe7040fa15/frontend/src/app/components/wp-custom-actions/wp-custom-actions/wp-custom-action.component.ts#L42"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

openprojectci commented on frontend/src/app/components/wp-edit/wp-edit-field/wp-replacement-label.component.ts:
```diff
@@ -27,8 +27,8 @@
-import {WorkPackageEditFieldGroupComponent} from 'core-components/wp-edit/wp-edit-field/wp-edit-field-group.directive';
 import {Component, ElementRef, Input, OnInit} from '@angular/core';
+import {EditFormComponent} from "core-app/modules/fields/edit/edit-form/edit-form.component";
 
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/370b05bb31fbf6a37ab5f5ec99ceabbe7040fa15/frontend/src/app/components/wp-edit/wp-edit-field/wp-replacement-label.component.ts#L32"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

openprojectci commented on frontend/src/app/components/wp-form-group/wp-attribute-group.component.ts:
```diff
@@ -28,9 +28,9 @@
 import {FieldDescriptor, GroupDescriptor} from 'core-components/work-packages/wp-single-view/wp-single-view.component';
 import {WorkPackageResource} from 'core-app/modules/hal/resources/work-package-resource';
+import {EditFormComponent} from "core-app/modules/fields/edit/edit-form/edit-form.component";
 
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/370b05bb31fbf6a37ab5f5ec99ceabbe7040fa15/frontend/src/app/components/wp-form-group/wp-attribute-group.component.ts#L34"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

openprojectci commented on frontend/src/app/components/wp-inline-create/wp-inline-create.component.ts:
```diff
@@ -58,6 +57,7 @@ import {WorkPackageInlineCreateService} from "core-components/wp-inline-create/w
 import {WorkPackageViewColumnsService} from "core-app/modules/work_packages/routing/wp-view-base/view-services/wp-view-columns.service";
 import {WorkPackageChangeset} from "core-components/wp-edit/work-package-changeset";
+import {EditForm} from "core-app/modules/fields/edit/edit-form/edit-form";
 
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/370b05bb31fbf6a37ab5f5ec99ceabbe7040fa15/frontend/src/app/components/wp-inline-create/wp-inline-create.component.ts#L61"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

openprojectci commented on frontend/src/app/components/wp-relations/embedded/children/wp-children-query.component.ts:
```diff
@@ -43,7 +43,7 @@ import {WorkPackageCacheService} from "core-components/work-packages/work-packag
 import {GroupDescriptor} from "core-components/work-packages/wp-single-view/wp-single-view.component";
-import {WorkPackageEventsService} from "core-app/modules/work_packages/events/work-package-events.service";
+import {HalEventsService} from "core-app/modules/hal/services/hal-events.service";
 
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/370b05bb31fbf6a37ab5f5ec99ceabbe7040fa15/frontend/src/app/components/wp-relations/embedded/children/wp-children-query.component.ts#L47"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

openprojectci commented on frontend/src/app/components/wp-relations/embedded/inline/add-existing/wp-relation-inline-add-existing.component.ts:
```diff
@@ -31,15 +31,16 @@ import {I18nService} from 'core-app/modules/common/i18n/i18n.service';
-import {WorkPackageEventsService} from "core-app/modules/work_packages/events/work-package-events.service";
+import {HalEventsService} from "core-app/modules/hal/services/hal-events.service";
+import {WorkPackageNotificationService} from "core-app/modules/work_packages/notifications/work-package-notification.service";
 
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/370b05bb31fbf6a37ab5f5ec99ceabbe7040fa15/frontend/src/app/components/wp-relations/embedded/inline/add-existing/wp-relation-inline-add-existing.component.ts#L44"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

openprojectci commented on frontend/src/app/components/wp-relations/embedded/relations/wp-relation-query.component.ts:
```diff
@@ -37,12 +37,12 @@ import {WorkPackageInlineCreateService} from "core-components/wp-inline-create/w
-import {WorkPackageEventsService} from "core-app/modules/work_packages/events/work-package-events.service";
+import {HalEventsService} from "core-app/modules/hal/services/hal-events.service";
+import {WorkPackageNotificationService} from "core-app/modules/work_packages/notifications/work-package-notification.service";
 
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/370b05bb31fbf6a37ab5f5ec99ceabbe7040fa15/frontend/src/app/components/wp-relations/embedded/relations/wp-relation-query.component.ts#L46"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

openprojectci commented on frontend/src/app/components/wp-relations/wp-relation-row/wp-relation-row.component.ts:
```diff
@@ -1,13 +1,14 @@
+import {HalEventsService} from "core-app/modules/hal/services/hal-events.service";
+import {WorkPackageNotificationService} from "core-app/modules/work_packages/notifications/work-package-notification.service";
 
 
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/370b05bb31fbf6a37ab5f5ec99ceabbe7040fa15/frontend/src/app/components/wp-relations/wp-relation-row/wp-relation-row.component.ts#L13"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

openprojectci commented on frontend/src/app/components/wp-relations/wp-relations-create/wp-relations-autocomplete/wp-relations-autocomplete.component.ts:
```diff
@@ -49,6 +49,7 @@ import {CurrentProjectService} from "core-components/projects/current-project.se
 import {HalResourceService} from "core-app/modules/hal/services/hal-resource.service";
 import {SchemaCacheService} from "core-components/schemas/schema-cache.service";
+import {WorkPackageNotificationService} from "core-app/modules/work_packages/notifications/work-package-notification.service";
 
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/370b05bb31fbf6a37ab5f5ec99ceabbe7040fa15/frontend/src/app/components/wp-relations/wp-relations-create/wp-relations-autocomplete/wp-relations-autocomplete.component.ts#L53"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

openprojectci commented on frontend/src/app/components/wp-relations/wp-relations-create/wp-relations-create.component.ts:
```diff
@@ -1,11 +1,11 @@
-import {WorkPackageEventsService} from "core-app/modules/work_packages/events/work-package-events.service";
+import {HalEventsService} from "core-app/modules/hal/services/hal-events.service";
+import {WorkPackageNotificationService} from "core-app/modules/work_packages/notifications/work-package-notification.service";
 
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/370b05bb31fbf6a37ab5f5ec99ceabbe7040fa15/frontend/src/app/components/wp-relations/wp-relations-create/wp-relations-create.component.ts#L9"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

openprojectci commented on frontend/src/app/components/wp-table/configuration-modal/wp-table-configuration.modal.ts:
```diff
@@ -16,18 +20,19 @@ import {OpModalComponent} from 'core-components/op-modals/op-modal.component';
+import {WorkPackageNotificationService} from "core-app/modules/work_packages/notifications/work-package-notification.service";
 
 export const WpTableConfigurationModalPrependToken = new InjectionToken<ComponentType<any>>('WpTableConfigurationModalPrependComponent');
 
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/370b05bb31fbf6a37ab5f5ec99ceabbe7040fa15/frontend/src/app/components/wp-table/configuration-modal/wp-table-configuration.modal.ts#L38"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

openprojectci commented on frontend/src/app/components/wp-table/timeline/container/wp-timeline-container.directive.ts:
```diff
@@ -50,15 +50,16 @@ import {WorkPackageTimelineCellsRenderer} from "core-components/wp-table/timelin
-import {WorkPackageEventsService} from "core-app/modules/work_packages/events/work-package-events.service";
+import {HalEventsService} from "core-app/modules/hal/services/hal-events.service";
+import {WorkPackageNotificationService} from "core-app/modules/work_packages/notifications/work-package-notification.service";
 
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/370b05bb31fbf6a37ab5f5ec99ceabbe7040fa15/frontend/src/app/components/wp-table/timeline/container/wp-timeline-container.directive.ts#L63"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

openprojectci commented on frontend/src/app/modules/attachments/attachment-list/attachment-list-item.component.ts:
```diff
@@ -31,7 +31,7 @@ import {I18nService} from 'core-app/modules/common/i18n/i18n.service';
 import {States} from 'core-components/states.service';
-import {WorkPackageNotificationService} from "core-components/wp-edit/wp-notification.service";
+import {HalResourceNotificationService} from "core-app/modules/hal/services/hal-resource-notification.service";
 
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/370b05bb31fbf6a37ab5f5ec99ceabbe7040fa15/frontend/src/app/modules/attachments/attachment-list/attachment-list-item.component.ts#L35"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

openprojectci commented on frontend/src/app/modules/boards/board/inline-add/board-inline-add-autocompleter.component.ts:
```diff
@@ -50,6 +49,7 @@ import {ApiV3FilterBuilder} from "core-components/api/api-v3/api-v3-filter-build
 import {SchemaCacheService} from "core-components/schemas/schema-cache.service";
 import {WorkPackageCardDragAndDropService} from "core-components/wp-card-view/services/wp-card-drag-and-drop.service";
+import {WorkPackageNotificationService} from "core-app/modules/work_packages/notifications/work-package-notification.service";
 
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/370b05bb31fbf6a37ab5f5ec99ceabbe7040fa15/frontend/src/app/modules/boards/board/inline-add/board-inline-add-autocompleter.component.ts#L53"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

openprojectci commented on frontend/src/app/modules/boards/new-board-modal/new-board-modal.component.ts:
```diff
@@ -37,7 +37,7 @@ import {BoardService} from "core-app/modules/boards/board/board.service";
 import {LoadingIndicatorService} from "core-app/modules/common/loading-indicator/loading-indicator.service";
-import {WorkPackageNotificationService} from "core-components/wp-edit/wp-notification.service";
+import {HalResourceNotificationService} from "core-app/modules/hal/services/hal-resource-notification.service";
 
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/370b05bb31fbf6a37ab5f5ec99ceabbe7040fa15/frontend/src/app/modules/boards/new-board-modal/new-board-modal.component.ts#L41"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

openprojectci commented on frontend/src/app/modules/common/autocomplete/user-autocompleter.component.ts:
```diff
@@ -34,9 +34,10 @@ import {I18nService} from "core-app/modules/common/i18n/i18n.service";
 import {NgSelectComponent} from "@ng-select/ng-select";
 import {DynamicBootstrapper} from "core-app/globals/dynamic-bootstrapper";
+import {HalResource} from "core-app/modules/hal/resources/hal-resource";
 
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/370b05bb31fbf6a37ab5f5ec99ceabbe7040fa15/frontend/src/app/modules/common/autocomplete/user-autocompleter.component.ts#L41"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

openprojectci commented on frontend/src/app/modules/common/autocomplete/version-autocompleter.component.ts:
```diff
@@ -35,7 +35,7 @@ import {VersionResource} from "core-app/modules/hal/resources/version-resource";
 import {I18nService} from "core-app/modules/common/i18n/i18n.service";
-import {WorkPackageNotificationService} from "core-components/wp-edit/wp-notification.service";
+import {HalResourceNotificationService} from "core-app/modules/hal/services/hal-resource-notification.service";
 
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/370b05bb31fbf6a37ab5f5ec99ceabbe7040fa15/frontend/src/app/modules/common/autocomplete/version-autocompleter.component.ts#L39"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

openprojectci commented on frontend/src/app/modules/fields/edit/editing-portal/edit-form-portal.component.ts:
```diff
@@ -18,7 +18,6 @@ import {
-import {WorkPackageChangeset} from "core-components/wp-edit/work-package-changeset";
 import {ResourceChangeset} from "core-app/modules/fields/changeset/resource-changeset";
 import {HalResource} from "core-app/modules/hal/resources/hal-resource";
 
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/370b05bb31fbf6a37ab5f5ec99ceabbe7040fa15/frontend/src/app/modules/fields/edit/editing-portal/edit-form-portal.component.ts#L23"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

openprojectci commented on frontend/src/app/modules/fields/edit/field-types/select-edit-field.component.ts:
```diff
@@ -39,7 +39,7 @@ import {SelectAutocompleterRegisterService} from "app/modules/fields/edit/field-
-  $href:string | null;
+  $href:string|null;
 }
 
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/370b05bb31fbf6a37ab5f5ec99ceabbe7040fa15/frontend/src/app/modules/fields/edit/field-types/select-edit-field.component.ts#L44"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

openprojectci commented on frontend/src/app/modules/work-package-graphs/configuration-modal/wp-graph-configuration.modal.ts:
```diff
@@ -1,29 +1,33 @@
+import {WorkPackageNotificationService} from "core-app/modules/work_packages/notifications/work-package-notification.service";
 
 export const WpTableConfigurationModalPrependToken = new InjectionToken<ComponentType<any>>('WpTableConfigurationModalPrependComponent');
 
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/370b05bb31fbf6a37ab5f5ec99ceabbe7040fa15/frontend/src/app/modules/work-package-graphs/configuration-modal/wp-graph-configuration.modal.ts#L33"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

openprojectci commented on frontend/src/app/modules/work_packages/routing/wp-base/wp--base.component.ts:
```diff
@@ -28,6 +28,10 @@
+import {WorkPackageEditFormRoutingService} from "core-app/modules/work_packages/routing/wp-edit-form/wp-edit-form-routing.service";
 
 export const wpBaseSelector = 'work-packages-base';
 
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/370b05bb31fbf6a37ab5f5ec99ceabbe7040fa15/frontend/src/app/modules/work_packages/routing/wp-base/wp--base.component.ts#L37"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

openprojectci commented:
<!--
  0 Errors
  14 Warnings: <a href="https://github.com/op...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          14 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/370b05bb31fbf6a37ab5f5ec99ceabbe7040fa15/frontend/src/app/components/modals/save-modal/save-query.modal.ts#L41">frontend/src/app/components/modals/save-modal/save-query.modal.ts#L41</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/370b05bb31fbf6a37ab5f5ec99ceabbe7040fa15/frontend/src/app/components/modals/share-modal/query-sharing.modal.ts#L41">frontend/src/app/components/modals/share-modal/query-sharing.modal.ts#L41</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/370b05bb31fbf6a37ab5f5ec99ceabbe7040fa15/frontend/src/app/components/wp-buttons/wp-status-button/wp-status-button.component.ts#L39">frontend/src/app/components/wp-buttons/wp-status-button/wp-status-button.component.ts#L39</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/370b05bb31fbf6a37ab5f5ec99ceabbe7040fa15/frontend/src/app/components/wp-details/wp-details-toolbar.component.ts#L33">frontend/src/app/components/wp-details/wp-details-toolbar.component.ts#L33</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/370b05bb31fbf6a37ab5f5ec99ceabbe7040fa15/frontend/src/app/modules/boards/board/board-list/board-list.component.ts#L57">frontend/src/app/modules/boards/board/board-list/board-list.component.ts#L57</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/370b05bb31fbf6a37ab5f5ec99ceabbe7040fa15/frontend/src/app/modules/boards/board/board.component.ts#L34">frontend/src/app/modules/boards/board/board.component.ts#L34</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/370b05bb31fbf6a37ab5f5ec99ceabbe7040fa15/frontend/src/app/modules/fields/edit/field-types/boolean-edit-field.component.ts#L32">frontend/src/app/modules/fields/edit/field-types/boolean-edit-field.component.ts#L32</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/370b05bb31fbf6a37ab5f5ec99ceabbe7040fa15/frontend/src/app/modules/fields/edit/field-types/date-edit-field.component.ts#L33">frontend/src/app/modules/fields/edit/field-types/date-edit-field.component.ts#L33</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/370b05bb31fbf6a37ab5f5ec99ceabbe7040fa15/frontend/src/app/modules/fields/edit/field-types/duration-edit-field.component.ts#L33">frontend/src/app/modules/fields/edit/field-types/duration-edit-field.component.ts#L33</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/370b05bb31fbf6a37ab5f5ec99ceabbe7040fa15/frontend/src/app/modules/fields/edit/field-types/float-edit-field.component.ts#L30">frontend/src/app/modules/fields/edit/field-types/float-edit-field.component.ts#L30</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/370b05bb31fbf6a37ab5f5ec99ceabbe7040fa15/frontend/src/app/modules/fields/edit/field-types/integer-edit-field.component.ts#L30">frontend/src/app/modules/fields/edit/field-types/integer-edit-field.component.ts#L30</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/370b05bb31fbf6a37ab5f5ec99ceabbe7040fa15/frontend/src/app/modules/fields/edit/field-types/multi-select-edit-field.component.ts#L37">frontend/src/app/modules/fields/edit/field-types/multi-select-edit-field.component.ts#L37</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/370b05bb31fbf6a37ab5f5ec99ceabbe7040fa15/frontend/src/app/modules/fields/edit/field-types/text-edit-field.component.ts#L31">frontend/src/app/modules/fields/edit/field-types/text-edit-field.component.ts#L31</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/370b05bb31fbf6a37ab5f5ec99ceabbe7040fa15/frontend/src/app/modules/fields/edit/field-types/work-package-edit-field.component.ts#L31">frontend/src/app/modules/fields/edit/field-types/work-package-edit-field.component.ts#L31</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

oliverguenther committed:
Remove another todo in halResourceNotification service

oliverguenther committed:
Fix npm TestBed for changed dependencies

HDinger committed:
Show inplace edit field in project details widget

oliverguenther committed:
Fix highlighting in single view

HDinger committed:
Provide HalResourceEditingService outside of project context

oliverguenther committed:
Used typedState for single-view

oliverguenther committed:
Also provide wpNotification service in split view

HDinger committed:
Correct check for resource type in eventsService

oliverguenther committed:
Fix getSchemaName in display field renderer

oliverguenther committed:
Fix passing ids into `halEditing.stopEditing`

oliverguenther committed:
Do not globally inject the halResourceEditingService

There's a bug(?) in ui-router that gives you the global service before
the parent injected service for a ui-view

oliverguenther committed:
Fix wpCreate service on copying and parallel creation

HDinger committed:
Remove test project widget

HDinger committed:
Revert changes for project details widget
